main.py

```diff
@@ -535,19 +536,32 @@ def load_evcs(self, event):
             for circuit_id in self._circuits_by_interface.get(interface_id,
                                                               []):
                 if circuit_id in circuits and circuit_id not in self.circuits:
-                    try:
-                        evc = self._evc_from_dict(circuits[circuit_id])
-                    except ValueError as exception:
-                        log.error(
-                            f'Could not load EVC {circuit_id} '
-                            f'because {exception}')
-                        continue
-
-                    evc.deactivate()
-                    evc.current_path = Path([])
-                    evc.sync()
-                    self.circuits.setdefault(circuit_id, evc)
-                    self.sched.add(evc)
+                    self._load_evc(circuits[circuit_id])
+
+    def load_all_evcs(self):
+        """Try to load all EVCs on startup."""
+        for circuit_id, circuit in self.storehouse.get_data().items():
+            if circuit_id not in self.circuits:
+                self._load_evc(circuit)
+
+    def _load_evc(self, circuit_dict):
+        """Load one EVC from storehouse to memory."""
+        try:
+            evc = self._evc_from_dict(circuit_dict)
+        except ValueError as exception:
+            log.error(
+                f'Could not load EVC {circuit_dict["id"]} '
```

Comment:
Maybe in a really rare situation, when there's no "id", this line can lead to an error